Sources/Commands/SwiftTestTool.swift

```diff
@@ -288,6 +287,56 @@ public class SwiftTestTool: SwiftTool<TestToolOptions> {
                 executionStatus = .failure
             }
         }
+
+        if options.shouldEnableCodeCoverage {
+            let buildParameters = try self.buildParameters()
+            let fs = Basic.localFileSystem
+            mergeRawDataFiles(fs: fs, path: buildParameters.buildPath.appending(component: "codecov"))
+            if buildParameters.triple.isDarwin() {
+                let graph = try loadPackageGraph()
+                let component = graph.rootPackages[0].name + "PackageTests"
+                let testPath = try buildTestsIfNeeded(options, graph: graph).appending(component: "Contents").appending(component: "MacOS").appending(components: component)
+                generateReport(fs: fs, testPath: testPath, buildPath: buildParameters.buildPath)
+            }
+        }
+
+        if options.showCodeCoverage {
+            let fs = Basic.localFileSystem
+            let buildParameters = try self.buildParameters()
+            do {
+                let output = try fs.readFileContents(buildParameters.buildPath.appending(component: "codecov").appending(component: "jsonData.txt"))
+                if let outputString = output.asString {
+                    print(outputString)
+                }
+            } catch {
+                print("Error while displaying code coverage results \(error.localizedDescription)")
```

Comment:
I think we should get rid of these individual do/catch statements in each function and instead making them throwing.